### PR TITLE
Show progress for taking/applying backups

### PR DESCRIPTION
**Pull request by ggezer**
Fixes https://github.com/Alluxio/alluxio/issues/8979

### CODE REVIEW

File: shell/src/main/java/alluxio/cli/fsadmin/command/BackupCommand.java [hunk 1]
```diff
@@ -63,12 +65,21 @@ public int run(CommandLine cl) throws IOException {
       opts.setTargetDirectory(args[0]);
     }
     opts.setLocalFileSystem(cl.hasOption(LOCAL_OPTION.getLongOpt()));
-    BackupResponse resp = mMetaClient.backup(opts.build());
+    // Create progress thread for showing progress while backup is in progress.
```
Comment (LuQQiu) [inline shell/src/main/java/alluxio/cli/fsadmin/command/BackupCommand.java]: Currently the detailed backup progress is shown in leader master.log, looks like we are not able to print the detailed backup progress to users directly. Can we add some user-facing console message to tell them the detailed backup progress is in leader master master.log?

Comment (ggezer) [reply shell/src/main/java/alluxio/cli/fsadmin/command/BackupCommand.java]: We can, but details we can provide is not very interesting for a shell user. We cannot show percentage or estimation. We can only show how many entries applied so far, which is not indicative of anything, that's why we decided to go with simple dot-dot progress.

File: core/client/fs/src/main/java/alluxio/client/meta/MetaMasterClient.java [hunk 2]
```diff
@@ -35,7 +35,7 @@
    * @param options backup options
    * @return the server response
    */
-  BackupResponse backup(BackupPOptions options) throws IOException;
+  BackupPResponse backup(BackupPOptions options) throws IOException;
```
Comment (gpang) [inline core/client/fs/src/main/java/alluxio/client/meta/MetaMasterClient.java]: will this break client compatibility?

Comment (ggezer) [reply core/client/fs/src/main/java/alluxio/client/meta/MetaMasterClient.java]: Yes.

Comment (gpang) [reply core/client/fs/src/main/java/alluxio/client/meta/MetaMasterClient.java]: @apc999 are we ok with this client change?

**Comment by ggezer** [comment]
Could someone please merge this PR? 
@gpang @apc999.

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**